syntax = "proto3";

package speaker_change_detection;

// Speaker Change Detection service definition.
service SpeakerChangeDetection {
  rpc Analyze (Request) returns (Response) {}
}

message Request {
  repeated float signal = 1 [packed=true];
  int32 sample_rate = 2;
}

message Changes {
  repeated float time = 1;
}

message Response {
  Changes changes = 1;
  float exec_time = 2;
}
